extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-preview.js

```diff
@@ -0,0 +1,751 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpStoryPlayer} from '../../../src/amp-story-player/amp-story-player-impl';
+import {
+  addAttributeAfterTimeout,
+  removeAfterTimeout,
+  updateHash,
+} from './utils';
+import {closest} from '../../../src/dom';
+import {escapeCssSelectorIdent} from '../../../src/css';
+import {htmlFor} from '../../../src/static-template';
+import {setStyles} from '../../../src/style';
+
+/**
+ * Creates a tab content, will be deleted when the tabs get implemented.
+ * @param {!Window} win
+ * @param {string} storyUrl
+ * @param {?string} devices
+ * @return {!Element} the layout
+ */
+export function createTabPreviewElement(win, storyUrl, devices) {
+  const element = win.document.createElement('amp-story-dev-tools-tab-preview');
+  element.setAttribute('data-story-url', storyUrl);
+  devices ? element.setAttribute('data-devices', devices) : null;
+  return element;
+}
+
+/**
+ * Generates the template for a device.
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildDeviceTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-dev-tools-device">
+      <div class="i-amphtml-story-dev-tools-device-screen">
+        <amp-story-player width="1" height="1" layout="container">
+          <a></a>
+        </amp-story-player>
+      </div>
+    </div>
+  `;
+};
+
+/**
+ * Generates the template for a device chip.
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildDeviceChipTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <button
+      class="i-amphtml-story-dev-tools-device-chip"
+      data-action="toggleDeviceChip"
+    >
+      <span class="i-amphtml-story-dev-tools-device-chip-name">Name</span>
+      <svg
+        title="cross"
+        xmlns="http://www.w3.org/2000/svg"
+        viewBox="0 0 24 24"
+        fill="white"
+        width="18px"
+        height="18px"
+      >
+        <rect width="20" height="3" x="2" y="10.5"></rect>
+        <rect
+          width="3"
+          height="20"
+          x="10.5"
+          y="2"
+          class="i-amphtml-story-dev-tools-device-chip-add-stick"
+        ></rect>
+      </svg>
+    </button>
+  `;
+};
+
+/**
+ * Generates the template for a help button.
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildHelpButtonTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <button
+      class="i-amphtml-story-dev-tools-button i-amphtml-story-dev-tools-button-help"
+    >
+      <span>HELP</span>
+      <svg
```

Comment:
Having the SVGs in the CSS might make all these templates a lot easier to read

Reply:
Will do!

Reply:
Moved most of the svgs into the CSS. The only one that is still in the html is the `+/-` icon besides the devices names because the rects are animated independently so we need the html elements for that.